### ISSUE TITLE
Debugger line highlight color is not colourblind friendly

### ISSUE THREAD

**Issue by krassowski** [jupyterlab/jupyterlab#10388]
## Description

The line highlight in the debugger makes it very difficult to read the line for me. I propose to change the following line:

https://github.com/jupyterlab/jupyterlab/blob/f0790b3ffd187b7dc923156b8ae227855abecefb/packages/debugger/style/editor.css#L6-L8

## Reproduce

1. Set a breakpoint
2. Observe the "SOURCE" box of the debugger
3. Step into a containing CodeMirror -highlighted code with various color
3. See the highlight being poor contrast

Example code:

```python
def test(x):
    def test_func(): 'test' if isinstance(x, str) else 123
```

### Light theme (current)

Most of the text is difficult to read.

![Screenshot from 2021-06-11 14-22-16](https://user-images.githubusercontent.com/5832902/121692978-935fe900-cac0-11eb-862d-eb737f24e674.png)

## Dark theme (current)

This is completely unreadable for me.

![Screenshot from 2021-06-11 14-22-30](https://user-images.githubusercontent.com/5832902/121692922-804d1900-cac0-11eb-93bf-5d53508a534f.png)

## Expected behavior

Please vote on proposals in the comments.

## Context

- Browser and version: Chrome
- JupyterLab version: 3.1.0a12

**Comment by krassowski** [comment]
## Option 1: `warn-color1`, without text shadow

This is still not readable for me.

```css
.jp-DebuggerEditor-highlight {
    background-color: var(--jp-warn-color1);
}
```

Light:

![Screenshot from 2021-06-11 14-25-05](https://user-images.githubusercontent.com/5832902/121693222-d5892a80-cac0-11eb-9aae-0196f9569564.png)

Dark:

![Screenshot from 2021-06-11 14-24-27](https://user-images.githubusercontent.com/5832902/121693148-c30ef100-cac0-11eb-849f-7576d3473ba6.png)

**Comment by krassowski** [comment]
## Option 2: `warn-color1`, with subtle text shadow

Light mode seems ok-ish, but the shadow does not help in the dark mode 

```css
.jp-DebuggerEditor-highlight {
    background-color: var(--jp-warn-color1);
    text-shadow: 0 0 10px var(--jp-layout-color1);
}
```

Light:
![Screenshot from 2021-06-11 14-26-58](https://user-images.githubusercontent.com/5832902/121693497-1da84d00-cac1-11eb-805c-10a19d51f0cb.png)

Dark:
![Screenshot from 2021-06-11 14-27-30](https://user-images.githubusercontent.com/5832902/121693569-2f89f000-cac1-11eb-93af-580b0fdbe47c.png)

**Comment by krassowski** [comment]
### 3. Using separate color for light and dark theme: brown

- light: `var(--md-brow-100)`
- dark: `var(--md-brow-800)`

Light:

![Screenshot from 2021-06-11 14-37-39](https://user-images.githubusercontent.com/5832902/121695077-a1166e00-cac2-11eb-84f0-40b246d2b28c.png)

Dark:

![Screenshot from 2021-06-11 14-36-05](https://user-images.githubusercontent.com/5832902/121694820-614f8680-cac2-11eb-9b6d-79fd6d01a9b5.png)

**Comment by krassowski** [comment]
### 3. Using separate color for light and dark theme: teal

- light: `var(--md-teal-100)`
- dark: `var(--md-teal-800)`

Light:

![Screenshot from 2021-06-11 14-38-44](https://user-images.githubusercontent.com/5832902/121695208-bf7c6980-cac2-11eb-83dd-a5ca882f1a90.png)

Dark:

![Screenshot from 2021-06-11 14-39-29](https://user-images.githubusercontent.com/5832902/121695291-d91db100-cac2-11eb-980f-5f0aa90c3864.png)

**Comment by krassowski** [comment]
### 3. Using separate color for light and dark theme: brown + shadow

- light: `var(--md-brown-100)`
- dark: `var(--md-brown-800)`

with subtle text-shadow to boost contrast `text-shadow: 0 0 5px var(--jp-layout-color1);`

Light:

![Screenshot from 2021-06-11 14-41-37](https://user-images.githubusercontent.com/5832902/121695655-2863e180-cac3-11eb-89f1-fa454fe25099.png)

Dark:

![Screenshot from 2021-06-11 14-41-04](https://user-images.githubusercontent.com/5832902/121695572-141fe480-cac3-11eb-8976-2e4a26cda11d.png)

**Comment by krassowski** [comment]
@isabela-pf if you got time I would very much like to hear you opinion on this. Feel free to suggest a different set of colours.

**Comment by isabela-pf** [comment]
Thanks for making this issue and proposing so many solutions! I'll give them a try as soon as possible.

**Comment by krassowski** [comment]
Note: the debugger in Chrome dev console alleviates the trade-off of having less pronounced background in two ways:
- by adding an arrow-like blue pointer on the line numbering gutter
- by adding a 1px border of more pronounced colour around the highlighted line:

![Screenshot from 2021-06-15 23-09-50](https://user-images.githubusercontent.com/5832902/122130006-e0302080-ce2e-11eb-8d57-77b6cca8c702.png)

Not that the have a perfect contrast, but these ideas seem nice.

**Comment by isabela-pf** [comment]
Okay! First, here's my thoughts on the solutions you've proposed. I agree that Option 5 seems the most legible! My main concern is that it seems difficult to spot because better legibility for the text means it doesn't stand out much from the background (this is a normal problem for situations like this). I'd also prefer to use colors already in JupyterLab, though using Material Design colors is already better than totally random ones in this case.

I took a little time to come up with another proposal based on your line number highlighting suggestion and trying to increase the contrast of the highlighted text.


Light
- Line number has background of `--jp-brand-color1` and the text and breakpoints are `--jp-inverse-font-color1`
- Background for the rest is `--jp-layout-color1` at 50% opacity
<img width="470" alt="Light 0-10" src="https://user-images.githubusercontent.com/50221806/122591096-54c5b380-d017-11eb-8cca-6cb6ec55c9ab.png">

Dark
- Line number has background of `--jp-brand-color0` and the text and breakpoints are `--jp-font-color1`
- Background for the rest is `--jp-layout-color1` at 50% opacity
<img width="466" alt="Dark 0-10" src="https://user-images.githubusercontent.com/50221806/122591137-63ac6600-d017-11eb-8f6f-784fb5d2454e.png">

Let me know your thoughts! I don't know if this is even feasible to make happen.

**Comment by krassowski** [comment]
Thank you. I would like to avoid the blue bit on the gutter as it signifies breakpoints in other debuggers, not the active line (I initially mixed these up - sorry!).

The proposed colours look good! What about the 1px outline/border above and below the highlighted line? I saw this in another editor recently too.

**Comment by isabela-pf** [comment]
That makes sense. Without using the blue in the gutter, I do think it makes sense to have the outline/border on the highlighted line since the background colors are not very dark. Were you thinking 100% opacity versions of the colors next to them, or something else?

**Comment by krassowski** [comment]
I guess that some value between 60% and 100% opacity would work well. I have not tried it yet, open to suggestions (and away from usable computer, my main laptop has crashed).

**Comment by krassowski** [comment]
Sorry I missed this earlier but `--jp-layout-color1` does not work for inactive cells (and we need to consider both active an inactive cells) as inactive cells already have `--jp-layout-color1`.
